src/main/python/envbuilder/cli.py

```diff
@@ -0,0 +1,276 @@
+
+                    for v in all_versions:
+                        if re.search(regex, v):
+                            version = v
+                            break
+                else:
+                    version = all_versions[0]
+
+                if version is not None:
+                    if current_version != version:
+                        updated_versions[env_file_version_key] = f"{current_version} --> {version}"
+                else:
+                    print(f"Unable to find version for {dep}")
+
+        return updated_versions
+
+    @staticmethod
+    def read_env_file(path='./.env'):
+        envfile = []
+        with open(path) as f:
+            for line in f:
+                envfile.append(line)
+        return envfile
+
+    @staticmethod
+    def env_to_dict(envfile):
+        compose_env_vars = {}
+        for line in envfile:
+            line = line.strip()
+            if '=' in line:
+                (key, val) = line.split('=', 1)
+                compose_env_vars[key] = val
+        return compose_env_vars
+
+    @staticmethod
+    def write_env_file(envfile, custom_versions, outfile):
+        '''Merge and write env file'''
+        with open(outfile, 'w') as env_file:
+            # rewrite the file with comments
+            for line in envfile:
+                if '=' in line:
+                    (key, val) = line.split('=', 1)
+                    s = f"{key}={custom_versions[key]}"
+                    print(s)
+                    env_file.write(f"{s}\n")
+                else:
+                    print(line, end='')
+                    env_file.write(line)
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+        description='docker image tag lookup')
+
+    parser.add_argument('--repo', required=False, default=None,
+                        help=('The repo to lookup tags'))
+
+    parser.add_argument('--org', required=False, default='edgexfoundry',
+                        help=('The organization to lookup'))
+
+    parser.add_argument('--envfile', required=False, default='./.env',
+                        help=('compose env file to read'))
+
+    parser.add_argument('--out', required=False, default='./.env.new',
+                        help=('where to write the generated env file'))
+
+    parser.add_argument('--no-lookup', action='store_true',
+                        help='do not lookup edgex versions')
+
+    parser.add_argument('--verbose', action='store_true',
+                        help='verbose logging')
+
+    # these two go together
+    parser.add_argument('--no-deps', action='store_true',
+                        help='do not lookup dependency versions')
+    parser.add_argument('--deps', required=False, default='vault consul redis kong kuiper mosquitto',
+                        help=('dependency versions to lookup'))
+
+    return parser.parse_args()
+
+
+def set_logging(args):
```

Comment:
logging messages to a log file can/will help debugging, especially if wanting to do multi-processing down the line:
you can also log all messages to a file using a file handler:
https://github.com/soda480/github-contributions/blob/main/src/main/python/ghcontrib/ghcontrib.py#L68-L80